pkg/controller/buildrun/buildrun_controller.go

```diff
@@ -159,7 +159,24 @@ func (r *ReconcileBuildRun) Reconcile(request reconcile.Request) (reconcile.Resu
 	if lastTaskRun != nil {
 		// TODO: Make this safer
 		if len(lastTaskRun.Status.Conditions) > 0 {
+			buildRunStatus := buildRun.Status.Succeeded
 			taskRunStatus := lastTaskRun.Status.Conditions[0].Status
+
+			// check if we should delete the generated service account by checking the build run spec
+			// and that the build run status has not yet the completion state from the task run
+			if isGeneratedServiceAccountUsed(buildRun) &&
+				(buildRunStatus == "" || buildRunStatus == corev1.ConditionUnknown) &&
+				(taskRunStatus != "" && taskRunStatus != corev1.ConditionUnknown) {
+				serviceAccount := &corev1.ServiceAccount{}
+				serviceAccount.Name = getGeneratedServiceAccountName(buildRun)
+				serviceAccount.Namespace = buildRun.Namespace
+
+				if err = r.client.Delete(ctx, serviceAccount); err != nil {
+					ctxlog.Error(ctx, err, "Error during deletion of generated service account.")
+					// Not failing here as there is also an owner reference = the build run deletes the service account
```

Comment:
Is it better that we ignore and don't reconcile this error?

Reply:
That's the place as I am seeking opinions for as mentioned in the description. https://github.com/redhat-developer/build/pull/265#issue-432333622. So, what is your opinion @zhangtbj @qu1queee @otaviof ?

Reply:
spoke with @SaschaSchwarze0 on this, we should return the error if the error was because deleting failed due to SA not there:
```go
				if err = r.client.Delete(ctx, serviceAccount); err != nil && !apierrors.IsNotFound(err) {
					ctxlog.Error(ctx, err, "Error during deletion of generated service account.")
					return reconcile.Result{}, err
				}
```